lib/purescript-ast/src/Control/Monad/Supply.hs

```diff
@@ -1,19 +1,20 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE PackageImports #-}
 
 -- |
 -- Fresh variable supply
 --
 module Control.Monad.Supply where
```

Comment:
This module and `Control.Monad.Supply` are moving to `purescript-ast` because they are used (transitively) in `Language.PureScript.Names`. My understanding is that the only reason this module exists in this repo in the first place is that there wasn't a fresh variable supply module to pull in. Were we to depend on a fresh variable supply module, we would assume it to be a further down dependency than `purescript-ast`: `purescript` -> `purescript-ast` -> `monad-supply`, or whatever.

More than down to flip some dependencies if that's not right. Wanted to call out why this module moved.